privaterelay/views.py

```diff
@@ -130,10 +130,20 @@ def waitlist(request):
     if not email or not fxa_uid:
         return JsonResponse({}, status=400)
 
-    _, created = Invitations.objects.get_or_create(
-        fxa_uid=fxa_uid, email=email, active=False
-    )
-    status = 201 if created else 200
+    try:
+        invitation =  Invitations.objects.get(
+            Q(fxa_uid=fxa_uid) | Q(email=email), active=False
+        )
+        if not invitation.fxa_uid:
+            invitation.fxa_uid = fxa_uid
+            invitation.save()
+        if invitation.email != email:
+            invitation.email = email
+            invitation.save()
```

Comment:
Not sure if we need this. If we already have a email change event from Firefox Accounts, we should have a separate event listener for that?

Reply:
Interesting. Shouldn't happen, but this logic makes sense too and will fix anything that might be messed up, right?

Reply:
We talked about keeping this in here as is and adding a TODO to get events from Firefox Accounts and update emails on the invitation as the profile update is already being observed for users.

Reply:
Opened https://github.com/mozilla/fx-private-relay/issues/378 for this.